src/Main.cpp

```diff
@@ -1,10 +1,223 @@
 #include <iostream>
-#include <genset_whisperpower_ddc/Dummy.hpp>
+#include <list>
+#include <memory>
+#include <genset_whisperpower_ddc/VariableSpeedMaster.hpp>
+#include <genset_whisperpower_ddc/GeneratorStatus.hpp>
+#include <genset_whisperpower_ddc/ControlCommand.hpp>
+#include <bits/stdc++.h>
+
+using namespace std;
+using namespace genset_whisperpower_ddc;
+
+#define STATUS_CASE(name) case STATUS_##name: return #name;
+std::string statusToString(GeneratorStatus status) {
+    switch (status) {
+        STATUS_CASE(NONE)
+        STATUS_CASE(PRE_GLOW)
+        STATUS_CASE(GLOW)
+        STATUS_CASE(START)
+        STATUS_CASE(SWITCH)
+        STATUS_CASE(STOP)
+        STATUS_CASE(FAILURE)
+        STATUS_CASE(PRESENT)
+        STATUS_CASE(FIRST)
+        STATUS_CASE(ERROR_BYPASS)
+        STATUS_CASE(PRE_ERROR_BYPASS)
+        STATUS_CASE(PUMPING)
+        STATUS_CASE(RESTART)
+        STATUS_CASE(HYDROBOOST)
+        STATUS_CASE(UNKNOWN)
+    }
+    return "UNKNOWN";
+}
+
+void usage(ostream& stream) {
+    stream << "usage: genset_whisperpower_ddc_ctl URI CMD\n"
+           << "where:\n"
+           << "  URI is a iodrivers_base URI\n"
+           << "\n"
+           << "Available Commands\n"
+           << "  send_control_command CONTROL_COMMAND: send a control command to the DDC GenVerter power controller\n"
+           << "    CONTROL_COMMAND can be start, stop or keep_alive\n"
+           << "  read_frame: read a frame and print its content\n"
+           << endl;
+}
 
 int main(int argc, char** argv)
 {
-    genset_whisperpower_ddc::DummyClass dummyClass;
-    dummyClass.welcome();
+    list<string> args(argv + 1, argv + argc);
+    if (args.size() < 2) {
+        bool error = !args.empty();
+        usage(error ? cerr : cout);
+        return error;
+    }
+    string uri = args.front();
+    args.pop_front();
+
+    if (args.empty()) {
+        usage(cerr);
+        return 1;
+    }
+
+    string cmd = args.front();
+    args.pop_front();
+
+    std::unique_ptr<genset_whisperpower_ddc::VariableSpeedMaster> genset_whisperpower_ddc_master;
+    auto* master = new genset_whisperpower_ddc::VariableSpeedMaster();
+    genset_whisperpower_ddc_master.reset(master);
```

Comment:
First, since you have (rightly) imported the namespace, don't put it explicitely

Second,

~~~
unique_ptr uptr;
auto* ptr = new ...
uptr.reset(ptr);
~~~

is just a very verbose way to do 

~~~
unique_ptr uptr(new ...);
~~~

I would also recommend renaming the unique ptr to `master`.